lineman/app/components/change_password_form/change_password_form.coffee

```diff
@@ -0,0 +1,17 @@
+angular.module('loomioApp').factory 'ChangePasswordForm', ->
+  templateUrl: 'generated/components/change_password_form/change_password_form.html'
+  controller: ($scope, $rootScope, CurrentUser, Records, FlashService) ->
+    $scope.user = CurrentUser.clone()
+
+    $scope.submit = ->
+      $scope.isDisabled = true
+      Records.users.changePassword($scope.user).then ->
+        $scope.isDisabled = false
+        FlashService.success('profile_page.messages.password_changed')
+        $scope.$close()
+      , (response) ->
+        $scope.isDisabled = false
+        if response.status == 422
```

Comment:
I'm liking this pattern quite a bit - if the error status is unprocessable entity, add validation errors, otherwise broadcast it to the app.

This means we can universally handle 500 (something went wrong), 400 (bad request), or 404 (connection dropped?) in the same place.